addons/sale_project/models/sale_order_line.py

```diff
@@ -37,7 +37,7 @@ def default_get(self, fields):
                 except AccessError:
                     del res['order_id']
 
-            if 'order_id' in fields and 'order_id' not in res:
+            if 'order_id' in fields and ('order_id' not in res or not res.get('order_id')):
```

Comment:
What's the point of this change?

Reply:
This whole condition check if order_id not in fields or present in fields and if either the key 'order_id' is missing in the res dictionary or the value of 'order_id' in res is false

Reply:
I did some testing and I now see why it is necessary. However, since we want to check if the value is False, the condition is a bit redundant, and you could just keep this:
```suggestion
            if 'order_id' in fields and not res.get('order_id'):
```